PicoProducer/python/analysis/ModuleTauPair.py

```diff
@@ -428,13 +428,19 @@ def fillMETAndDiLeptonBranches(self, event, tau1, tau2, met, met_vars):
     
     # PROPAGATE TES/LTF/JTF shift to MET (assume shift is already applied to object)
     if self.ismc and 't' in self.channel:
-      if hasattr(tau1,'es') and tau1.es!=1:
-        dp = tau1.tlv*(1.-1./tau1.es) # assume shift is already applied
-        correctmet(met,dp)
-      if hasattr(tau2,'es') and tau2.es!=1:
-        dp = tau2.tlv*(1.-1./tau2.es)
-        #print ">>> fillMETAndDiLeptonBranches: Correcting MET for es=%.3f, pt=%.3f, dpt=%.3f, gm=%d"%(tau2.es,tau2.pt,dp.Pt(),tau2.genPartFlav)
-        correctmet(met,tau2.tlv*(1.-1./tau2.es))
+      try: 
+         if hasattr(tau1,'es') and tau1.es!=1:
```

Comment:
@cardinia reported a similar problem. It's not clear to me why this stopped working, because it seems to work for me in CMSSW 12.4.8... What python & CMSSW version are you using?

Any case, it might be better to use this solution,
```
if tau1.__dict__.get('es',1)!=1:
```
https://github.com/cms-tau-pog/TauFW/pull/48/files#diff-2d8b00f0e02ba9192427efa300462bbaa9cc5a83a12283c729499e02bcc45f73R431
without the `try`/`except` and instead of the `hassattr`. The `es` will probably not be defined at least once event, unless you're using the di-tau module, and printing something out every event (in the except) will worsen the performance significantly.

This PR will anyway conflict with PR https://github.com/cms-tau-pog/TauFW/pull/48.

Reply:
I am using `CMSSW_12_4_8` and `Python 3.9.6`.
Your solution works fine, thanks